src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -1,330 +1,390 @@
+                    {
+                        if (links == null)
+                        {
+                            links = new List<Activity>(messages.Length);
+                        }
+
+                        links.Add(link);
+                    }
+                }
+
+                if (links != null)
+                {
+                    return new Dictionary<string, object> {["Links"] = links};
+                }
+
+                return EmptyScope;
+            }
+
+            private bool TryGetLinkedActivity(EventData message, out Activity link)
+            {
+                link = null;
+
+                if (((message.SystemProperties != null && message.SystemProperties.TryGetValue("Diagnostic-Id", out var diagnosticIdObj)) || message.Properties.TryGetValue("Diagnostic-Id", out diagnosticIdObj)) 
+                    && diagnosticIdObj is string diagnosticIdString)
+                {
+                    link = new Activity("Microsoft.Azure.EventHubs.Process");
+                    link.SetParentId(diagnosticIdString);
+                    return true;
+                }
+
+                return false;
+            }
+
+            private string GetOperationDetails(PartitionContext context, string operation, IDictionary<string, string> triggerDetails = null)
+            {
+                JObject contextJO = new JObject()
+                {
+                    { "partitionId", context.PartitionId },
+                    { "owner", context.Owner },
+                    { "eventHubPath", context.EventHubPath }
+                };
+
+                JObject rootJO = new JObject()
+                {
+                    { "operation", operation },
+                    { "partitionContext", contextJO }
+                };
+
+                if (triggerDetails != null)
+                {
+                    rootJO["triggerDetails"] = JObject.FromObject(triggerDetails);
+                }
+
+                // Log partition lease
+                if (context.Lease != null)
+                {
+                    contextJO["lease"] = new JObject()
+                    {
+                        { "offset", context.Lease.Offset },
+                        { "sequenceNumber", context.Lease.SequenceNumber}
+                    };
+                }
+
+                // Log RuntimeInformation if EnableReceiverRuntimeMetric is enabled
+                if (context.RuntimeInformation != null)
+                {
+                    contextJO["runtimeInformation"] = new JObject()
+                    {
+                        { "lastEnqueuedOffset", context.RuntimeInformation.LastEnqueuedOffset },
+                        { "lastSequenceNumber", context.RuntimeInformation.LastSequenceNumber},
+                        { "lastEnqueuedTimeUtc", context.RuntimeInformation.LastEnqueuedTimeUtc}
+                    };
+                }
+
+                return rootJO.ToString();
+            }

```

Comment:
This will likely be called a lot, right? You may want to do something like I did here -- https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs#L72 and use a JSONTextWriter directly.

I never did any measuring -- but I read that this was the fastest way to write out JSON using Newtonsoft...